### PR TITLE
Helpful Shell Extensions

### LINKED ISSUE #13861: Add Shell Extensions Helpers
Allowing to replace for example

    if (_shellSettings.State == TenantState.Running)
    ...
    var settings = new ShellSettings()
    {
        Name = ShellHelper.DefaultShellName,
        State = TenantState.Running
    };

By

    if (_shellSettings.IsRunning())
    ...
    var settings = new ShellSettings()
        .AsDefaultShell()
        .AsRunning();

And so on

### PR DESCRIPTION
**Pull request by jtkech**
Allowing to replace for example

    var modelUrlPrefix = model.RequestUrlPrefix?.Trim(' ', '/') ?? String.Empty;

    var modelUrlHosts = model.RequestUrlHost
        ?.Split(_hostSeparators, StringSplitOptions.RemoveEmptyEntries)
        ?? Array.Empty<string>();

    if (_shellHost.GetAllSettings().Any(settings =>
        settings != existingShellSettings &&
        String.Equals(settings.RequestUrlPrefix ?? String.Empty, modelUrlPrefix, StringComparison.OrdinalIgnoreCase) &&
        DoesUrlHostExist(settings.RequestUrlHosts, modelUrlHosts)))

    ...

    if (settings.State != TenantState.Uninitialized &&
        (settings.State != TenantState.Disabled ||
        _shellHost.TryGetShellContext(settings.Name, out var value) && value.ActiveScopes > 0))

By

    if (_shellHost.GetAllSettings().Any(settings =>
        settings != existingShellSettings &&
        settings.HasUrlPrefix(model.RequestUrlPrefix) &&
        settings.HasUrlHost(model.RequestUrlHost)))

    ...

    if (!settings.IsRemovable() || _shellHost.IsShellActive(settings))

Include some null checks making the usage easier.

All to simplifiy tenant state management.

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Abstractions/Shell/Extensions/ShellSettingsExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace OrchardCore.Environment.Shell
+{
+    public static class ShellSettingsExtensions
+    {
+        /// <summary>
+        /// Wether it is the 'Default' tenant or not.
+        /// </summary>
+        public static bool IsDefaultShell(this ShellSettings settings) =>
+            settings is not null && settings.Name == ShellHelper.DefaultShellName;
```
Comment (kevinchalet) [inline src/OrchardCore/OrchardCore.Abstractions/Shell/Extensions/ShellSettingsExtensions.cs:11]: You need moar pattern matching love in your life 😃 

```suggestion
            settings is { Name: ShellHelper.DefaultShellName };
```

File: src/OrchardCore/OrchardCore.Abstractions/Shell/Extensions/ShellSettingsExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        /// Wether the tenant is disabled or not.
+        /// </summary>
+        public static bool IsDisabled(this ShellSettings settings) =>
+            settings is not null && settings.State == TenantState.Disabled;
+
+        /// <summary>
+        /// Wether the tenant is initialized or not.
+        /// </summary>
+        public static bool IsInitialized(this ShellSettings settings) =>
+            settings is not null && (settings.IsRunning() || settings.IsDisabled());
```
Comment (kevinchalet) [inline src/OrchardCore/OrchardCore.Abstractions/Shell/Extensions/ShellSettingsExtensions.cs:41]: Pattern matching everywhere! :trollface: 

```suggestion
            settings is { State: TenantState.Running or TenantState.Disabled };
```

File: src/OrchardCore/OrchardCore.Abstractions/Shell/Extensions/ShellSettingsExtensions.cs (hunk 1 above)
Comment (jtkech) [reply src/OrchardCore/OrchardCore.Abstractions/Shell/Extensions/ShellSettingsExtensions.cs:11]: Thanks for the suggestion, beautiful ;)

File: src/OrchardCore/OrchardCore.Abstractions/Shell/Extensions/ShellSettingsExtensions.cs (hunk 2 above)
Comment (jtkech) [reply src/OrchardCore/OrchardCore.Abstractions/Shell/Extensions/ShellSettingsExtensions.cs:41]: Beautiful!

**Comment by jtkech** [comment]
I think it could have been useful but the related issue has been back logged.

Hmm from memory maybe it was also fixing a couple of minor issues but not sure.

**Comment by jtkech** [comment]
Re-open as I think they are helpful extensions.

**Pull request merged**